src/core.jl

```diff
@@ -74,9 +74,19 @@ appropriately defined for child types.
 """
 AbstractFilter
 
+# In clojure a reduction function is one of signature
+# whatever, input -> whatever
+#
+# The `Reduction` type corresponds to such a function, but keeps extra information:
+# * InType records the type of input
+# * `xform` and `inner` are a decomposition of the reduction function into
+#   a reducer `xform` and an inner reduction function `inner`.
+#   `inner` reduction. `inner` can be either a `Reduction` or a function with
+#   arity-2 and arity-1 methods
```

Comment:
This not is certainly helpful to me, but it may or may not be helpful to other people. Should we keep it?

Reply:
This is great. Let's keep it! We need a docstring for `Reduction` at some point. This is a good the starting point.